### ISSUE TITLE
cl65 writes object files into current work directory by default

### ISSUE THREAD

**Issue by mrdudz** [cc65/cc65#168]
what the topic says. they should go into the system temp directory instead

**Comment by greg-king5** [comment]
Actually, cl65 allows the other tools to do their default action of putting their output files next to their input files.  Only the final output file can be put into a different location.  Usually, that's what we want:  all related files with the same name (and different suffixes) are grouped together.

But occasionally, we want to avoid cluttering out input directory with any output files.  Therefore, we need a cl65 option that will put intermediate files into a separately named directory -- something like
`--temp-dir <dir>`.

I see this issue as a feature request, instead of a bug report.

**Comment by oliverschmidt** [comment]
I disagree that this is usually what we want. Rather I don't see why any write operation should be performed in the directory containing the input/source file(s).

Anyway I agree with you to classify this issue as an enhancement (aka feature request) rather than a bug.

**Comment by MonteCarlos** [comment]
If someone would like to add the object file to a library with ar65 he 
needs them in the source directory or subsequent directories.
This is what every ide does, i think.

MC


Am 04.07.2015 um 22:34 schrieb Oliver Schmidt:

**Comment by mrdudz** [comment]
On Tuesday 07 July 2015, 04:42:11 MonteCarlos <notifications@github.com> 
wrote:
> If someone would like to add the object file to a library with ar65 he
> needs them in the source directory or subsequent directories.
> This is what every ide does, i think.

in that case you would explicitly instruct cl65 to produce an object file as 
output though, and _then_ it will go to the specified directory, of course.

-- 

http://www.hitmen-console.org    http://magicdisk.untergrund.net
http://www.pokefinder.org        http://ar.pokefinder.org

Aus Fehlern wird man klug. Darum ist einer nicht genug.

**Comment by fo-fo** [comment]
I agree with @oliverschmidt and @mrdudz that system temporary directory would be preferable.

Related issue is that if you have two files, say `test.c` and `test.s` and you compile them with `cl65 test.c test.s`, cl65 ends up overwriting and deleting `test.s`. This actually happened to me a while ago.

**Comment by MonteCarlos** [comment]
thats why .c.o or .s.o is seen in makefiles very often


Am 24.03.2016 um 10:40 schrieb fo-fo:

**Comment by spiro-trikaliotis** [comment]
Hello,

* On Thu, Mar 24, 2016 at 02:40:31AM -0700 fo-fo wrote:
 
> Related issue is that if you have two files, say test.c and test.s and you
> compile them with cl65 test.c test.s, cl65 ends up overwriting and deleting
> test.s. This actually happened to me a while ago.

Other compilers do the same thing. Try gcc -S test.c and have fun seeing
your test.s vanish. ;)

Regards,
Spiro.

-- 
Spiro R. Trikaliotis
http://www.trikaliotis.net/

**Comment by fo-fo** [comment]
> Other compilers do the same thing. Try gcc -S test.c and have fun seeing
> your test.s vanish. ;)

It's a different scenario though: in that case you're explicitly asking for `test.s` to be generated. If you do `gcc test.s test.c` both files are preserved, as expected.

**Comment by MonteCarlos** [comment]
Thay why people use .s.o and .c.o.


Am 26.03.2016 um 03:26 schrieb fo-fo:

**Comment by Grandmother** [comment]
I have the related issue. I tried to use cc65 with cmake and it has troubles while performing some compiler checks. The root of trouble is that cl65 try to write temporary files to directory where source file is, but this directory in my case is cmake's '/usr/blah/blah' directory. So it just hasn't permissions to write to this directory.

That's why output directory or system temporary directory is preferable -- cl65 has permissions to write there.

The other reason is to keep source directory clean. This is the rule of cmake and I agree with this rule as well as probably all other cmake users.

**Comment by GorillaSapiens** [comment]
is this bug still valid after #2660?  can we just close this one out?

**Comment by mrdudz** [comment]
should be solved
